File: AiFront/app/src/main/java/com/aifront/data/model/Conversation.kt
package com.aifront.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "conversations")
data class Conversation(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val title: String,
    val modelId: String? = null,
    val modelName: String? = null,
    val vendorName: String? = null,
    val conversationType: ConversationType = ConversationType.TEXT,
    val isClusterMode: Boolean = false,
    val clusterModelIds: String? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(),
    val messageCount: Int = 0
)

enum class ConversationType {
    TEXT,
    CODE,
    IMAGE,
    CLUSTER
}

File: ai_model_aggregator/lib/models/conversation.dart
class Conversation {
  final String id;
  final String title;
  final String modelId;
  final String modelName;
  final String vendorName;
  final String type; // text / code / image / cluster
  final DateTime createdAt;
  final DateTime updatedAt;
  final List<String>? clusterModelIds;

  const Conversation({
    required this.id,
    required this.title,
    required this.modelId,
    required this.modelName,
    required this.vendorName,
    required this.type,
    required this.createdAt,
    required this.updatedAt,
    this.clusterModelIds,
  });

  Map<String, dynamic> toMap() => {
        'id': id,
        'title': title,
        'modelId': modelId,
        'modelName': modelName,
        'vendorName': vendorName,
        'type': type,
        'createdAt': createdAt.toIso8601String(),
        'updatedAt': updatedAt.toIso8601String(),
        'clusterModelIds': clusterModelIds?.join(',') ?? '',
      };

  factory Conversation.fromMap(Map<String, dynamic> map) => Conversation(
        id: map['id'] as String,
        title: map['title'] as String,
        modelId: map['modelId'] as String,
        modelName: map['modelName'] as String,
        vendorName: map['vendorName'] as String,
        type: map['type'] as String,
        createdAt: DateTime.parse(map['createdAt'] as String),
        updatedAt: DateTime.parse(map['updatedAt'] as String),
        clusterModelIds: (map['clusterModelIds'] as String?)?.isNotEmpty == true
            ? (map['clusterModelIds'] as String).split(',')
            : null,
      );
}